app/controllers/concerns/checkout_callbacks.rb

```diff
@@ -15,7 +15,10 @@ module CheckoutCallbacks
     prepend_before_action :require_distributor_chosen
 
     before_action :load_order, :associate_user, :load_saved_addresses, :load_saved_credit_cards
-    before_action :load_shipping_methods, if: -> { params[:step] == "details" }
+    before_action :load_shipping_methods,
+                  :load_allowed_shipping_methods, if: -> {
+                                                        params[:step] == "details"
+                                                      }
```

Comment:
If we adjust the view to only work with allowed methods then we don't need to load and store all methods separately.

Reply:
I needed to rename `load_allowed_shipping_methods` to `allowed_shipping_methods`. This is to avoid the linter issue 
```ruby
app/controllers/concerns/checkout_callbacks.rb:52:5: C: Naming/MemoizedInstanceVariableName: Memoized variable @allowed_shipping_methods does not match method name load_allowed_shipping_methods. Use @load_allowed_shipping_methods instead.
    @allowed_shipping_methods ||= sorted_available_shipping_methods.filter(
    ^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```